### PR TITLE
15 ✅ feat: add CertificateStore implementations for all backends

### PR DESCRIPTION
**Pull request by Gozala**
## Summary

The access layer needs to persist delegation certificates so authorization
chains can be verified across sessions. Each backend stores encoded
certificate bytes keyed by audience/subject/issuer: FileSystem as files,
Volatile in a HashMap, IndexedDb using IDBKeyRange prefix queries.

Tests live in dialog-ucan which provides the concrete Ucan protocol type.
open_store() returns the platform backend (FileSystem on native, IndexedDb
on wasm) so the same assertions run everywhere. 20 tests cover the full
prove flow including powerline chains, time bounds, and edge cases.

### CODE REVIEW

File: rust/dialog-storage/src/storage/provider/fs/certificate.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+#[async_trait::async_trait]
+impl<P: Protocol> CertificateStore<P> for FileSystem
+where
+    P::Certificate: Send + Sync,
+{
+    /// List certificates where `audience` is the recipient and `subject`
+    /// is either the specific subject DID or, when `None`, a
+    /// [powerline](https://github.com/ucan-wg/delegation?tab=readme-ov-file#powerline)
```
Comment (cdata) [inline rust/dialog-storage/src/storage/provider/fs/certificate.rs:30]: Given that a "powerline" delegation may be seen as a more powerful scope than specifying some subject, I wonder if it would be more defensive to make the subject a domain-specific enum (rather than `Option<...>`).

File: rust/dialog-storage/src/storage/provider/volatile/certificate.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    ///
+    /// Each certificate is stored keyed by
+    /// `{audience}/{subject}/{issuer}.{hash}` where the hash is the
+    /// base58-encoded BLAKE3 hash of the encoded certificate bytes.
+    async fn save(&self, delegation: &P::Delegation) -> Result<(), AuthorizeError> {
+        let mut sessions = self.sessions.write();
+
+        for cert in delegation.certificates() {
+            let bytes = cert.encode()?;
+            let id = base58::ToBase58::to_base58(blake3::hash(&bytes).as_bytes().as_slice());
```
Comment (cdata) [inline rust/dialog-storage/src/storage/provider/volatile/certificate.rs:62]: You should be able to `use base58::ToBase58` above and then invoke this as a method on any slice of `u8`

File: rust/dialog-ucan/src/certificate_store.rs [hunk 3]
```diff
@@ -0,0 +1,556 @@
+#[cfg(test)]
+mod tests {
+    #[cfg(target_arch = "wasm32")]
+    wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_dedicated_worker);
+
+    use dialog_capability::access::{Certificate, CertificateStore};
+    use dialog_credentials::Ed25519Signer;
+    use dialog_effects::storage::{Directory, Location};
+    use dialog_storage::resource::Resource;
```
Comment (cdata) [inline rust/dialog-ucan/src/certificate_store.rs:9]: IIUC this constitutes a circular dependency (via `dev-dependencies`, so technically not a problem) between `dialog-ucan` and `dialog-storage`. There may not be a better way, but I tend to try to avoid these kinds of cycles even via `dev-dependencies`.

**Review by cdata — APPROVED** [review]
LGTM

File: rust/dialog-ucan/src/certificate_store.rs (hunk 3 above)
Comment (Gozala) [reply rust/dialog-ucan/src/certificate_store.rs:9]: I think only real way to break the cycle here is create separate crate that depends on `dialog-ucan` and `dialog-storage` in order to test interop. crate just for tests seemed like a too much, so it kind of ended up with this admittedly annoying situation.

Perhaps you have recommendation for addressing this kind of situation ?

File: rust/dialog-storage/src/storage/provider/volatile/certificate.rs (hunk 2 above)
Comment (Gozala) [reply rust/dialog-storage/src/storage/provider/volatile/certificate.rs]: fixed

**Comment by Gozala** [comment]
Merged into `main` as part of squash commit a898b5d ("feat: Capability system, repository, and query integration"). Closing as merged.
